### PR TITLE
feat: add casdoor springboot starter

### PR DESCRIPTION
**Pull request by tangyang9464**
Signed-off-by: tangyang9464 <tangyang9464@163.com>

close: #1

### CODE REVIEW

File: src/test/resources/test.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+casdoor.endpoint=https://www.baidu.com
```
Comment (hsluoyz) [inline src/test/resources/test.properties:1]: What's this?

File: src/main/java/org/casbin/casdoor/CasdoorSpringBootStarterApplication.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.casbin.casdoor;
```
Comment (hsluoyz) [inline src/main/java/org/casbin/casdoor/CasdoorSpringBootStarterApplication.java:1]: Casbin license header

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: src/test/resources/test.properties (hunk 1 above)
Comment (tangyang9464) [reply src/test/resources/test.properties]: modified

File: src/main/java/org/casbin/casdoor/CasdoorSpringBootStarterApplication.java (hunk 2 above)
Comment (tangyang9464) [reply src/main/java/org/casbin/casdoor/CasdoorSpringBootStarterApplication.java]: deleted

**Comment by hsluoyz** [comment]
@Kininaru @sh1luo plz review

**Review by hsluoyz — APPROVED** [review]

**Review by sh1luo — APPROVED** [review]
Looks very standardized, no problem

**Comment by hsluoyz** [comment]
@sh1luo can you setup a Java web application to use this starter?

**Comment by sh1luo** [comment]
> @sh1luo can you setup a Java SpringBoot application to use this starter as login?

ok

**Comment by hsluoyz** [comment]
@fangzhengjin plz review

File: src/main/java/org/casbin/casdoor/config/CasdoorAutoConfigure.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import javax.annotation.Resource;
+
+/**
+ * @author tangyang9464
+ */
+@Configuration
+@EnableConfigurationProperties(CasdoorConfiguration.class)
+public class CasdoorAutoConfigure {
+    @Resource
+    private CasdoorConfiguration properties;
```
Comment (fangzhengjin) [inline src/main/java/org/casbin/casdoor/config/CasdoorAutoConfigure.java:20]: For mandatory dependencies, it is recommended to use constructor injection.

**Review by fangzhengjin — CHANGES_REQUESTED** [review]

File: src/main/java/org/casbin/casdoor/config/CasdoorConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package org.casbin.casdoor.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author tangyang9464
+ */
+@ConfigurationProperties(prefix = "casdoor")
+@Component
```
Comment (fangzhengjin) [inline src/main/java/org/casbin/casdoor/config/CasdoorConfiguration.java:10]: Configuration classes specified with the @EnableConfigurationProperties annotation are automatically injected without the @Component annotation.

File: src/main/java/org/casbin/casdoor/config/CasdoorAutoConfigure.java (hunk 3 above)
Comment (tangyang9464) [reply src/main/java/org/casbin/casdoor/config/CasdoorAutoConfigure.java]: changed

**Review by fangzhengjin — APPROVED** [review]
LGTM

**Pull request merged**